File: solutions/c/isogram/1/isogram.c
#include "isogram.h"

bool is_isogram(const char phrase[]) {

    bool b;

    if      ( phrase    == NULL ) { b = false; }
    else if ( phrase[0] == '\0' ) { b = true;  }
    else {
        int size = 0; while ( phrase[size] != '\0' ) size++;
        char c[255] = {0};
        b = true;
        for ( int i = 0; i < size; i++ ) {
            if (phrase[i] != 32 && phrase[i] != 45 && phrase[i] != 46 ) {
                int ascii = tolower(phrase[i]);
                c[ascii]++;
                if ( c[ascii] > 1 ) {
                    b = false;
                    break;
                }
            }
        }
    }

    return b;
}

File: solutions/lua/isogram/1/isogram.lua
local ascii_range = { 65, 90 };

function is_isogram(phrase)
    local boolean = true

    local counter = {}
    for i = 1, string.len(phrase) do
        local upper = string.upper(phrase)
        local char  = string.sub(upper, i, i)
        local byte  = string.byte(char)
        if byte >= ascii_range[1] and byte <= ascii_range[2] then
            counter[byte] = counter[byte] and counter[byte] + 1 or 1
            if counter[byte] > 1 then boolean = false end
        end
    end
    
    return boolean
end

return is_isogram
